x-pack/plugins/apm/public/hooks/use_service_name.tsx

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useContext } from 'react';
+import { useParams } from 'react-router-dom';
+import { ServiceNameContext } from '../context/service_name_context';
+
+/**
+ * Get the current service name, or undefined if there is no current service.
+ *
+ * Attempts to use the path parameters and then the ServiceNameContext provider.
+ *
+ * There are some situations (like the action menu -> alert triggers) where we
+ * don't have the full route context but we can supply the service name through
+ * a separate context.
```

Comment:
This is probably fine but want to make sure I understand:

I don't follow why we need context for accessing service name when we should be able to read it from the url. Is it because it's not always in the url? Or is service name in the url but we can't access it with `useParams` for a technical reason (where `window.location.search` still works I assume)?

Reply:
While can read it through the URL (`window.location.pathname`), we don't have the full router context with all the routes (we pretty much only have the `history`), we don't have all the `Route`s defined. `useParams` matches against those routes so doesn't work.

We may actually be able to [`useRouteMatch`](https://reactrouter.com/web/api/Hooks/useroutematch) to do this without having to use a context, so I might try that in a subsequent PR.

Reply:
Sounds good 👍